tools/assets-automation/assets-maintenance-tool/Azure.Sdk.Tools.Assets.MaintenanceTool/Scan/AssetsScanner.cs

```diff
@@ -0,0 +1,357 @@
+
+        public AssetsScanner(string workingDirectory) {
+            WorkingDirectory = workingDirectory;
+        }
+
+        public AssetsScanner()
+        {
+            WorkingDirectory = Directory.GetCurrentDirectory();
+        }
+
+        /// <summary>
+        /// Walk a run configuration an create a resultSet of all found assets.json references.
+        /// 
+        /// This function automatically takes previous output into account by checking in the current
+        /// working directory for an "output.json" file.
+        /// </summary>
+        /// <param name="config"></param>
+        /// <returns></returns>
+        public AssetsResultSet Scan(RunConfiguration config)
+        {
+            var resultSet = new List<AssetsResult>();
+            var existingResults = ParseExistingResults();
+
+            Parallel.ForEach(config.Repos, repoConfig =>
+            {
+                resultSet.AddRange(ScanRepo(repoConfig, existingResults));
+            });
+
+            var newResults = new AssetsResultSet(resultSet);
+
+            Save(newResults);
+
+            return newResults;
+        }
+
+        /// <summary>
+        /// If the tool is invoked in a directory containing an `"output.json" file, that file will be parsed
+        /// for it's results. The file itself is merely a List of type AssetsResult serialized to disk.
+        /// </summary>
+        /// <returns></returns>
+        public AssetsResultSet? ParseExistingResults()
+        {
+            if (File.Exists(ResultsFile))
+            {
+                using var stream = System.IO.File.OpenRead(ResultsFile);
+                using var doc = JsonDocument.Parse(stream);
+
+                var results = JsonSerializer.Deserialize<List<AssetsResult>>(doc);
+
+                if (results != null)
+                {
+                    return new AssetsResultSet(results);
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Given a repo configuration, scan the repo and return the assetsresults from all targeted branches.
+        /// </summary>
+        /// <param name="config"></param>
+        /// <param name="previousOutput"></param>
+        /// <returns></returns>
+        private List<AssetsResult> ScanRepo(RepoConfiguration config, AssetsResultSet? previousOutput)
+        {
+            string? envOverride = Environment.GetEnvironmentVariable(GIT_TOKEN_ENV_VAR);
+            var authString = string.Empty;
+            if (!string.IsNullOrWhiteSpace(envOverride))
+            {
+                authString = $"{envOverride}@";
+            }
+
+            var targetRepoUri = $"https://{authString}github.com/{config.Repo}.git";
+            var workingDirectory = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+            var results = new List<AssetsResult>();
+
+            try
+            {
+                if(!Directory.Exists(workingDirectory))
```

Comment:
```suggestion
                if (!Directory.Exists(workingDirectory))
```

In general, I like consistency when it comes to adding space after control structures like `if` or `foreach`. I see this problem in few more places.